### PR TITLE
REFACTOR only supervisor of course can update course, validate date, show user

### PR DESCRIPTION
**Pull request by phunghangbk**

### CODE REVIEW

File: app/controllers/admin/user_courses_controller.rb [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+  before_action :set_course, only: [:show]
+  before_action :super_of_course, only: [:show]
 
   def show
-    @course = Course.find params[:course_id]
+  end
+
+  private
+  def set_course
+  	@course = Course.find params[:course_id]
```
Comment (phonghenry14) [inline app/controllers/admin/user_courses_controller.rb]: indent

File: app/controllers/admin/user_courses_controller.rb [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+  end
+
+  def super_of_course
+    unless @course.supervisors.include?(current_user)
```
Comment (phonghenry14) [inline app/controllers/admin/user_courses_controller.rb]: try remove ( )

File: app/controllers/admin/courses_controller.rb [hunk 3]
```diff
@@ -52,4 +53,11 @@ def course_params
     params.require(:course).permit :name, :start, :description, :finish, :status, supervisor_ids: [],
       trainee_ids: [], subject_ids: [], course_subjects_attributes: [:status]
   end
+
+  def super_of_course
```
Comment (VoTaiTri) [inline app/controllers/admin/courses_controller.rb]: duplicate, try remove that to module and call it

Comment (phunghangbk) [inline app/controllers/admin/courses_controller.rb]: Sr anh, trong file kia em khong can dung ham nay ah!

Comment (phonghenry14) [inline app/controllers/admin/courses_controller.rb]: hu cau.:))

**Pull request merged**